File: Data Structures & Algorithms/jump-game/submission-3.cpp
class Solution{
public:
    bool canJump(vector<int>& nums){
        int n=nums.size();
        int fart=0;
        for(int i=0;i<n;i++){
            if(i>fart) return false;
            fart=max(fart,nums[i]+i);
        }
        return true;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution{
    public int longestConsecutive(int[] nums){
        HashSet<Integer> st=new HashSet<>();
        for(int i:nums) st.add(i);
        int currNum,currLen,longest=0;
        for(int x:st){
            if(!st.contains(x-1)){
                currNum=x;
                currLen=1;
                while(st.contains(currNum+1)){
                    currNum++;
                    currLen++;
                }
                longest=Math.max(longest,currLen);
            }
        }
        return longest;
    }
}